src/file.js

```diff
@@ -179,29 +179,39 @@ Sk.builtin.file.prototype["readlines"] = new Sk.builtin.func(function (self, siz
     return new Sk.builtin.list(arr);
 });
 
-Sk.builtin.file.prototype["seek"] = new Sk.builtin.func(function (self, offset, whence) {
+Sk.builtin.file.prototype["seek"] = new Sk.builtin.func(function seek(self, offset, whence) {
+    var l_offset =  Sk.ffi.remapToJs(offset);
+
     if (whence === undefined) {
-        whence = 1;
+        whence = 0;
     }
-    if (whence == 1) {
-        self.pos$ = offset;
-    } else {
-        self.pos$ = self.data$ + offset;
+    if (whence == 0) {
+        self.pos$ = l_offset;
+    } else if (whence == 1) {
+        self.pos$ = self.data$.length + l_offset;
+    } else if (whence == 2) {
+        self.pos$ = self.data$.length + l_offset;
     }
-});
 
-Sk.builtin.file.prototype["tell"] = new Sk.builtin.func(function (self) {
-    return self.pos$;
+    return Sk.builtin.none.none$;
 });
 
+Sk.builtin.file.prototype["tell"] = new Sk.builtin.func(function tell(self) {
+    return Sk.ffi.remapToPy(self.pos$);
+});
 
-Sk.builtin.file.prototype["truncate"] = new Sk.builtin.func(function (self, size) {
+Sk.builtin.file.prototype["truncate"] = new Sk.builtin.func(function truncate(self, size) {
     goog.asserts.fail();
 });
 
-Sk.builtin.file.prototype["write"] = new Sk.builtin.func(function (self, str) {
-    if (self.fileno === 1) {
-        Sk.output(Sk.ffi.remapToJs(str));
+Sk.builtin.file.prototype["write"] = new Sk.builtin.func(function write(self, str) {
+    var mode = Sk.ffi.remapToJs(self.mode);
+    if (mode === "w" || mode === "wb" || mode === "a" || mode === "ab") {
+        if (self.fileno === 1) {
+            Sk.output(Sk.ffi.remapToJs(str));
+        } else {
+            goog.asserts.fail();
```

Comment:
OK, sorry two parameters!   the trinket implementation accepts the file object and a string.  so internally it would be called `Sk.writeFile(this, str)`